integration-tests/src/main/java/io/druid/testing/ConfigFileConfigProvider.java

```diff
@@ -53,69 +55,94 @@ private void loadProperties(String configFile)
     ObjectMapper jsonMapper = new ObjectMapper();
     try {
       props = jsonMapper.readValue(
-          new File(configFile), new TypeReference<Map<String, String>>()
-      {
-      }
+        new File(configFile), new TypeReference<Map<String, String>>()
+        {
+        }
       );
     }
     catch (IOException ex) {
       throw new RuntimeException(ex);
     }
     // there might not be a router; we want routerHost to be null in that case
-    routerHost = props.get("router_host");
-    if (null != routerHost) {
-	routerHost += ":" + props.get("router_port");
+    routerUrl = props.get("router_url");
+    if (routerUrl == null) {
+      String routerHost = props.get("router_host");
+      if (null != routerHost) {
+        routerUrl = String.format("http://%s:%s", routerHost, props.get("router_port"));
+      }
+    }
+    brokerUrl = props.get("broker_url");
+    if (brokerUrl == null) {
+      brokerUrl = String.format("http://%s:%s", props.get("broker_host"), props.get("broker_port"));
```

Comment:
it seems that earlier, user provided both "host:port" in the host property itself... now port is to be provided separately?